README
Introduction
------------

This is a `Django CMS` plugin for news.

Installation
------------

First, you must install `Django CMS` and `photologue` which require `Django` 
and a few other libraries such as `PIL`. For full details, see the installation 
instructions for those packages.

Install ``cmsplugin-newsy`` to your environment with a tool such as `PIP`, 
`setuptools`, or `buildout`.

Add ``newsy`` to the ``INSTALLED_APPS`` list in your project's 
``settings.py`` and run the ``syncdb`` command on your ``manage.py``.

.. _Django: http://www.djangoproject.com/
.. _Django CMS: https://www.django-cms.org/
.. _photologue: http://code.google.com/p/django-photologue/
.. _PIL: http://www.pythonware.com/products/pil/
.. _PIP: http://www.pip-installer.org/
.. _setuptools: http://pypi.python.org/pypi/setuptools/
.. _buildout: http://pypi.python.org/pypi/zc.buildout/

What's Inside
-------------


HISTORY.rst
0.6.1 (2012/07/30)
------------------

* Fixed a typo bug in the item view

0.6 (2012/06/15)
----------------

* Added a tags view
* Added support for tags with dashes
* Added a CMS Attach Menu for menu/breadcrumb integration
* Added year/month/day values to context when viewing news items with those
  filters applied

0.5.1 (2012/06/04)
------------------

* Added a null log handler
* Switch Tag Field to a Text based field to allow for more tags per item

0.5 (2012/06/01)
----------------

* Fixed the bug when moving plugins between placeholders (Github issue #15)
* Fixed page title when editing a news item (Github issue #17)
* Site selection is now a multi-checkbox input (Github issue #18)
* Fixed reversion integration (Github issue #19)
* Added a redirect to handle a changed publication date with a unique slug
  (Github issue #16)

0.4.1 (2011/12/08)
------------------

* Added debug logging output

0.4 (2011/10/12)
----------------

* Added a latest news plugin

0.3 (2011/10/12)
----------------

* Switched to using the class based generic list view properly
* Added tag browsing support
* Added default pagination on list views of 30 (when using shortcut views)
* Added some new methods to the NewsItem model to make a few things easier
* Method for getting tag and date filters in generic view
* Added ampersands to allowed tag characters (urls.py)
* Reorganized form fields and increased the size of the title fields and tag field
* News item slugs must be lowercase
* Added the 0003 data migration to lowercase existing slugs
* Added get_next_published and get_previous_published methods to news item
* Added RSS feed for all published and for all tags (one feed per tag)

0.2 (2011/07/25)
----------------

* Added an archive view (GitHub issue #10)
* Added tags field to NewsItem model for admin integration (GitHub issue #7)
* Added tags field to NewsItem administration (GitHub issue #7)
* Added south migration for the tags field (GitHub issue #7)
* Added a news_plugins_media tag for rendering plugin media links (GitHub issue #8)
* Added a site_objects model manager to news item (GitHub issue #11)
* Modified list views to use the site_objects model manager (GitHub issue #11)
* Added permission wrapper around list view and news item view for unpublished 
  items (GitHub issue #12)

0.1 (2011/07/15)
----------------

* A very basic, working version for demonstration


newsy/migrations/0001_initial.py
# encoding: utf-8
import datetime
from south.db import db
from south.v2 import SchemaMigration
from django.db import models

class Migration(SchemaMigration):

    def forwards(self, orm):
        
        # Adding model 'NewsItemThumbnail'
        db.create_table('newsy_newsitem_thumbnail', (
            ('id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('image', self.gf('django.db.models.fields.files.ImageField')(max_length=100)),
            ('date_taken', self.gf('django.db.models.fields.DateTimeField')(null=True, blank=True)),
            ('view_count', self.gf('django.db.models.fields.PositiveIntegerField')(default=0)),
            ('crop_from', self.gf('django.db.models.fields.CharField')(default='center', max_length=10, blank=True)),
            ('effect', self.gf('django.db.models.fields.related.ForeignKey')(blank=True, related_name='newsitemthumbnail_related', null=True, to=orm['photologue.PhotoEffect'])),
            ('news_item', self.gf('django.db.models.fields.related.OneToOneField')(related_name='thumbnail', unique=True, to=orm['newsy.NewsItem'])),
        ))
        db.send_create_signal('newsy', ['NewsItemThumbnail'])

        # Adding model 'NewsItem'
        db.create_table('newsy_newsitem', (
            ('id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('title', self.gf('django.db.models.fields.CharField')(max_length=255)),
            ('short_title', self.gf('django.db.models.fields.CharField')(max_length=255, null=True, blank=True)),
            ('page_title', self.gf('django.db.models.fields.CharField')(max_length=255, null=True, blank=True)),
            ('template', self.gf('django.db.models.fields.CharField')(max_length=255)),
            ('slug', self.gf('django.db.models.fields.SlugField')(max_length=255, db_index=True)),
            ('description', self.gf('django.db.models.fields.TextField')(null=True, blank=True)),
            ('publication_date', self.gf('django.db.models.fields.DateTimeField')(db_index=True, null=True, blank=True)),
            ('published', self.gf('django.db.models.fields.BooleanField')(default=False, db_index=True)),
        ))
        db.send_create_signal('newsy', ['NewsItem'])

        # Adding M2M table for field sites on 'NewsItem'
        db.create_table('newsy_newsitem_sites', (
            ('id', models.AutoField(verbose_name='ID', primary_key=True, auto_created=True)),
            ('newsitem', models.ForeignKey(orm['newsy.newsitem'], null=False)),
            ('site', models.ForeignKey(orm['sites.site'], null=False))
        ))
        db.create_unique('newsy_newsitem_sites', ['newsitem_id', 'site_id'])

        # Adding M2M table for field placeholders on 'NewsItem'
        db.create_table('newsy_newsitem_placeholders', (
            ('id', models.AutoField(verbose_name='ID', primary_key=True, auto_created=True)),
            ('newsitem', models.ForeignKey(orm['newsy.newsitem'], null=False)),
            ('placeholder', models.ForeignKey(orm['cms.placeholder'], null=False))
        ))
        db.create_unique('newsy_newsitem_placeholders', ['newsitem_id', 'placeholder_id'])


    def backwards(self, orm):
        
        # Deleting model 'NewsItemThumbnail'
        db.delete_table('newsy_newsitem_thumbnail')

        # Deleting model 'NewsItem'
        db.delete_table('newsy_newsitem')

        # Removing M2M table for field sites on 'NewsItem'
        db.delete_table('newsy_newsitem_sites')

        # Removing M2M table for field placeholders on 'NewsItem'
        db.delete_table('newsy_newsitem_placeholders')


    models = {
        'cms.placeholder': {
            'Meta': {'object_name': 'Placeholder'},
            'default_width': ('django.db.models.fields.PositiveSmallIntegerField', [], {'null': 'True'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'slot': ('django.db.models.fields.CharField', [], {'max_length': '50', 'db_index': 'True'})
        },
        'newsy.newsitem': {
            'Meta': {'ordering': "['-publication_date']", 'object_name': 'NewsItem'},
            'description': ('django.db.models.fields.TextField', [], {'null': 'True', 'blank': 'True'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'page_title': ('django.db.models.fields.CharField', [], {'max_length': '255', 'null': 'True', 'blank': 'True'}),
            'placeholders': ('django.db.models.fields.related.ManyToManyField', [], {'to': "orm['cms.Placeholder']", 'symmetrical': 'False'}),
            'publication_date': ('django.db.models.fields.DateTimeField', [], {'db_index': 'True', 'null': 'True', 'blank': 'True'}),
            'published': ('django.db.models.fields.BooleanField', [], {'default': 'False', 'db_index': 'True'}),
            'short_title': ('django.db.models.fields.CharField', [], {'max_length': '255', 'null': 'True', 'blank': 'True'}),
            'sites': ('django.db.models.fields.related.ManyToManyField', [], {'to': "orm['sites.Site']", 'symmetrical': 'False'}),
            'slug': ('django.db.models.fields.SlugField', [], {'max_length': '255', 'db_index': 'True'}),
            'template': ('django.db.models.fields.CharField', [], {'max_length': '255'}),
            'title': ('django.db.models.fields.CharField', [], {'max_length': '255'})
        },
        'newsy.newsitemthumbnail': {
            'Meta': {'object_name': 'NewsItemThumbnail', 'db_table': "'newsy_newsitem_thumbnail'"},
            'crop_from': ('django.db.models.fields.CharField', [], {'default': "'center'", 'max_length': '10', 'blank': 'True'}),
            'date_taken': ('django.db.models.fields.DateTimeField', [], {'null': 'True', 'blank': 'True'}),
            'effect': ('django.db.models.fields.related.ForeignKey', [], {'blank': 'True', 'related_name': "'newsitemthumbnail_related'", 'null': 'True', 'to': "orm['photologue.PhotoEffect']"}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'image': ('django.db.models.fields.files.ImageField', [], {'max_length': '100'}),
            'news_item': ('django.db.models.fields.related.OneToOneField', [], {'related_name': "'thumbnail'", 'unique': 'True', 'to': "orm['newsy.NewsItem']"}),
            'view_count': ('django.db.models.fields.PositiveIntegerField', [], {'default': '0'})
        },
        'photologue.photoeffect': {
            'Meta': {'object_name': 'PhotoEffect'},
            'background_color': ('django.db.models.fields.CharField', [], {'default': "'#FFFFFF'", 'max_length': '7'}),
            'brightness': ('django.db.models.fields.FloatField', [], {'default': '1.0'}),
            'color': ('django.db.models.fields.FloatField', [], {'default': '1.0'}),
            'contrast': ('django.db.models.fields.FloatField', [], {'default': '1.0'}),
            'description': ('django.db.models.fields.TextField', [], {'blank': 'True'}),
            'filters': ('django.db.models.fields.CharField', [], {'max_length': '200', 'blank': 'True'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'name': ('django.db.models.fields.CharField', [], {'unique': 'True', 'max_length': '30'}),
            'reflection_size': ('django.db.models.fields.FloatField', [], {'default': '0'}),
            'reflection_strength': ('django.db.models.fields.FloatField', [], {'default': '0.59999999999999998'}),
            'sharpness': ('django.db.models.fields.FloatField', [], {'default': '1.0'}),
            'transpose_method': ('django.db.models.fields.CharField', [], {'max_length': '15', 'blank': 'True'})
        },
        'sites.site': {
            'Meta': {'ordering': "('domain',)", 'object_name': 'Site', 'db_table': "'django_site'"},
            'domain': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'name': ('django.db.models.fields.CharField', [], {'max_length': '50'})
        }
    }

    complete_apps = ['newsy']
